proposals/extend_pod_security_admission.md

```diff
@@ -72,114 +71,149 @@ For example, if a user wants to apply Restricted PSS to the selected namespaces
-           # +required
-         - controlName: Capabilities
-           images:
-           - ghcr.io/example/nginx:1.2.3
-           # +optional
-           restrictedField: spec.containers[*].securityContext.capabilities.add
-           # Any type
-           # +optional
-           values: # how to define an object?
-           - "KILL"
-           - "SETGID"
-           - "SETUID"
+        profile: restricted
+        # version must be a valid Kubernetes minor version,
+        # or `latest`.
+        version: v1.24
+        exclude:
+          # controlName is the Control defined in PSS
+          # +required
+          - controlName: Volumes
+            labels:
+              env: prod
 ```
 
-Note that securityContext is configured at container level, not pod, as the matching image selects a specific container. The pod creation will be rejected if there's any violation found in its container. For pod level checks, other match criteria such as labels could be used.
+### Policy Reports
 
 For a PolicyReport, `cpol.rules.validate.podSecurity.exclude.restrictedField` and `cpol.rules.validate.podSecurity.exclude.controlName` will be added to `polr.results.properties` to audit the result.
 
 ```yaml
 PolicyReport:
 results:
-- message: |- 
-    'validation error: The label `app.kubernetes.io/name` is required. 
-    Rule check-for-labels failed at path
-    /metadata/labels/app.kubernetes.io/name/'
-  policy: require-labels
-  resources:
-  - apiVersion: v1
-    kind: Pod
-    name: nginx
-    namespace: default
-    uid: 13fcb726-5597-4fcd-a708-ce8b558bd484
-  rule: check-for-labels
-  properties:
-    controlName: Capabilities
-    podSecurity: spec.containers[*].securityContext.capabilities.add
-  scored: true
-  status: fail
+  - message: |-
+      'validation error: The label `app.kubernetes.io/name` is required. 
+      Rule check-for-labels failed at path
+      /metadata/labels/app.kubernetes.io/name/'
+    policy: require-labels
+    resources:
+      - apiVersion: v1
+        kind: Pod
+        name: nginx
+        namespace: default
+        uid: 13fcb726-5597-4fcd-a708-ce8b558bd484
+    rule: check-for-labels
+    properties:
+      controlName: Capabilities
+      podSecurity: spec.containers[*].securityContext.capabilities.add
+    scored: true
+    status: fail
 ```
 
-### Exceptions for container security context
+### Fields outside container object
 
-* Host Namespaces: `spec.hostNetwork,spec.hostPID,spec.hostIPC`
-* Sysctls: `spec.securityContext.sysctls[*].name`
-* volumes: `spec.volumes`
-* Non-root groups: `spec.securityContext.supplementalGroups[*], spec.securityContext.fsGroup`
+- Host Namespaces: `spec.hostNetwork,spec.hostPID,spec.hostIPC`
+- Sysctls: `spec.securityContext.sysctls[*].name`
+- volumes: `spec.volumes`
+- Non-root groups: `spec.securityContext.supplementalGroups[*], spec.securityContext.fsGroup`
 
+For these pod level checks, we can use labels to select them.
```

Comment:
Do we have a sample policy for this?

Reply:
Yes [here](https://github.com/kyverno/KDP/pull/27/files#diff-6445fcf14a55d9c90d6e0b83d196721b466908fdfc41cb653d515994d3ad098eR151),

We should move this part just before the sample policy.